src/dotnet/Media.Service/LinkPreviewsBackend.cs

```diff
@@ -66,74 +66,10 @@ public class LinkPreviewsBackend(IServiceProvider services)
-                .FirstOrDefaultAsync(x => x.Id == id.Value, cancellationToken)
-                .ConfigureAwait(false);
-
-            if (dbLinkPreview != null && SystemNow - dbLinkPreview.ModifiedAt.ToMoment() < TimeSpan.FromDays(1))
-                return dbLinkPreview.ToModel();
-
-            var linkMeta = await Crawler.Crawl(url, cancellationToken).ConfigureAwait(false);
-            if (dbLinkPreview == null) {
-                var linkPreview = new LinkPreview {
-                    Id = LinkPreview.ComposeId(url),
-                    Version = VersionGenerator.NextVersion(),
-                    Url = url,
-                    Title = linkMeta.Title,
-                    Description = linkMeta.Description,
-                    PreviewMediaId = linkMeta.PreviewMediaId,
-                    CreatedAt = SystemNow,
-                    ModifiedAt = SystemNow,
-                };
-                if (!linkMeta.VideoMetadata.IsNone)
-                    linkPreview = linkPreview with {
-                        VideoSite = linkMeta.VideoMetadata.SiteName,
-                        VideoUrl = linkMeta.VideoMetadata.SecureUrl,
-                        VideoWidth = linkMeta.VideoMetadata.Width,
-                        VideoHeight = linkMeta.VideoMetadata.Height,
-                    };
-
-                dbLinkPreview = new DbLinkPreview(linkPreview);
-                dbContext.Add(dbLinkPreview);
-            }
-            else {
-                dbLinkPreview = await dbContext.LinkPreviews.ForNoKeyUpdate()
-                    .FirstOrDefaultAsync(x => x.Id == id.Value, cancellationToken)
-                    .ConfigureAwait(false);
-                var linkPreview = dbLinkPreview!.ToModel();
-                linkPreview = linkPreview with {
-                    PreviewMediaId = linkMeta.PreviewMediaId,
-                    Title = linkMeta.Title,
-                    Description = linkMeta.Description,
-                    ModifiedAt = SystemNow,
-                    Version = VersionGenerator.NextVersion(linkPreview.Version),
-                };
-                if (!linkMeta.VideoMetadata.IsNone)
-                    linkPreview = linkPreview with {
-                        VideoSite = linkMeta.VideoMetadata.SiteName,
-                        VideoUrl = linkMeta.VideoMetadata.SecureUrl,
-                        VideoWidth = linkMeta.VideoMetadata.Width,
-                        VideoHeight = linkMeta.VideoMetadata.Height,
-                    };
-                dbLinkPreview.UpdateFrom(linkPreview);
-            }
-
-            await dbContext.SaveChangesAsync(cancellationToken).ConfigureAwait(false);
-            context.Operation().Items.Set(true);
-            return dbLinkPreview.ToModel();
-        }
-        finally {
-            await MarkNotCrawling(id).ConfigureAwait(false);
-        }
+        var (wasRun, result) = await MeshLocks
+            .TryRun(ct => RefreshUnsafe(id, url, ct), ToRedisKey(id), cancellationToken)
```

Comment:
Probably it makes sense to let it complete - even if the mesh lock is gone.

Reply:
But overall, this is minor.

Reply:
done